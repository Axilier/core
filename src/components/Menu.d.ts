import { MenuProps } from '../Types';
declare const Menu: ({ forwardedRef, open, children }: MenuProps) => JSX.Element;
export default Menu;
