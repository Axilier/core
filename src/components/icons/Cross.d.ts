import { IconProps } from '../../Types';
export declare const Cross: ({ style, className, iconColor, }: IconProps) => JSX.Element;
