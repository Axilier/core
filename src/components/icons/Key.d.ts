import { IconProps } from '../../Types';
export declare const Key: ({ iconColor, style, className, }: IconProps) => JSX.Element;
