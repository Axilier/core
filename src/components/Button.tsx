import React, { useState } from 'react';
import { ButtonProps } from '../Types';
import styles from '../css/Button.module.css';

const Button = ({
    btnType = 'button',
    variant = 'contained',
    btnColor = 'primary',
    btnStyles = {},
    btnIconBackgroundColor = 'white',
    disabled = false,
    btnIcon,
    children,
    onClick,
    style,
    className,
    id,
}: ButtonProps): JSX.Element => {
    const [hovered, setHovered] = useState(false);
    const mainColour = () => {
        switch (btnColor) {
            case 'primary':
                return '#057AFF';
            case 'secondary':
                return 'black';
            default:
                return btnColor;
        }
    };
    const backgroundColour = () => {
        switch (variant) {
            case 'contained':
                return hovered ? 'white' : mainColour();
            case 'outlined':
                return hovered ? mainColour() : 'white';
            case 'text':
                return 'transparent';
            default:
                return hovered ? 'white' : mainColour();
        }
    };
    const border = () => {
        switch (variant) {
            case 'contained':
                return hovered
                    ? `solid 2px ${mainColour()}`
                    : 'solid 2px transparent';
            case 'outlined':
                return `solid 2px ${mainColour()}`;
            case 'text':
                return 'none';
            default:
                return hovered
                    ? `solid 2px ${mainColour()}`
                    : 'solid 2px transparent';
        }
    };
    const textColor = () => {
        switch (variant) {
            case 'contained':
                return hovered ? mainColour() : 'white';
            case 'outlined':
                return hovered ? 'white' : mainColour();
            case 'text':
                return mainColour();
            default:
                return hovered ? mainColour() : 'white';
        }
    };
    const padding = () =>
        children && children.toString().length > 14 ? '10px' : '5px';

    return (
        <div className={`${styles.buttonContainer} ${className}`} style={style}>
            <button
                id={id || 'button'}
                // eslint-disable-next-line react/button-has-type
                type={btnType || 'button'}
                onMouseEnter={() => setHovered(true)}
                onMouseLeave={() => setHovered(false)}
                className={styles.button}
                onClick={onClick}
                style={{
                    backgroundColor: backgroundColour(),
                    color: textColor(),
                    border: border(),
                    padding: variant === 'text' ? 'unset' : `${padding()} 20px`,
                    ...btnStyles,
                }}
            >
                {btnIcon ? (
                    <div
                        id={'icon-container'}
                        className={styles.buttonIcon}
                        style={{
                            backgroundColor: btnIconBackgroundColor || 'white',
                        }}
                    >
                        {btnIcon}
                    </div>
                ) : null}
                {children}
            </button>
            {disabled ? <span id={'filter'} className={styles.filter} /> : null}
        </div>
    );
};

export default Button;
