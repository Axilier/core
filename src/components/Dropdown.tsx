/** @format */

// /** @format */
//
// // @flow
// import * as React from 'react';
// import { useEffect, useState } from 'react';
// import { DropdownProps } from '../Types';
//
// const Dropdown = ({ open, onChange, style, className }: DropdownProps) => {
//     const [isOpen, setIsOpen] = useState(open);
//
//     useEffect(() => {
//         setIsOpen(open);
//     }, [open]);
//
//     return (
//         <svg
//             transform={isOpen ? 'rotate(90)' : 'none'}
//             className={`drop-down-icon ${className}`}
//             onClick={() => {
//                 if (!onChange) return;
//                 onChange();
//             }}
//             viewBox={'0 0 64 64'}
//             xmlns={'http://www.w3.org/2000/svg'}
//             fillRule={'evenodd'}
//             clipRule={'evenodd'}
//             strokeLinejoin={'round'}
//             strokeMiterlimit={2}
//             style={style}
//         >
//             <g transform={'matrix(2.90892 0 0 2.9091 -4174.32 -1000.725)'}>
//                 <circle cx={1446} cy={355} r={11} fill={'#ebebeb'} />
//                 <path
//                     d={
//                         'M1446 344c6.07 0 11 4.929 11 11s-4.93 11-11 11-11-4.929-11-11 4.93-11 11-11zm0 .174c5.98 0 10.83 4.851 10.83 10.826 0 5.975-4.85 10.826-10.83 10.826-5.97 0-10.83-4.851-10.83-10.826 0-5.975 4.86-10.826 10.83-10.826z'
//                     }
//                     fill={'#d3d3d3'}
//                 />
//             </g>
//             <path
//                 d={'M47.297 31.109L23.8 47.997V14.22l23.497 16.889z'}
//                 fill={'#d3d3d3'}
//             />
//         </svg>
//     );
// };
//
// export default Dropdown;
