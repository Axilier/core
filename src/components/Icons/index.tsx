/** @format */

export { Lock } from './Lock';
export { Room } from './Room';
export { Mail } from './Mail';
export { Key } from './Key';
export { Google } from './Google';
export { Github } from './Github';
export { Warning } from './Warning';
export { Import } from './Import';
export { Tick } from './Tick';
export { Cross } from './Cross';
export { Question } from './Question';
export { Issue } from './Issue';
export { Copyright } from './Copyright';
export { Connections } from './Connections';
export { SharedFiles } from './SharedFiles';
export { Favourites } from './Favourites';
export { Recent } from './Recent';
export { AllFiles } from './AllFiles';
export { Search } from './Search';
export { Dropdown } from './Dropdown';
