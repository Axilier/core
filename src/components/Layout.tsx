import React, { CSSProperties } from 'react';
import { LayoutProps } from '../Types';

const Layout = ({
    center = false,
    orientation = 'row',
    style = {},
    children,
    elemRef,
    ...elementProps
}: LayoutProps &
    React.DetailedHTMLProps<
        React.HTMLAttributes<HTMLDivElement>,
        HTMLDivElement
    >): JSX.Element => {
    const styles = () => {
        const basicStyles: CSSProperties = {
            display: 'flex',
            flexDirection: orientation,
            alignItems: 'center',
            justifyContent: 'center',
        };
        if (!center) {
            delete basicStyles.alignItems;
            delete basicStyles.justifyContent;
        }
        return basicStyles;
    };
    return (
        <div
            {...elementProps}
            ref={elemRef}
            style={{
                ...styles(),
                ...style,
            }}
        >
            {children}
        </div>
    );
};

export default Layout;
