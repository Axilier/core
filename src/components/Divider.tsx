import React from 'react';
import { DividerProps } from '../Types';
import Layout from './Layout';

const Divider = ({ children, style }: DividerProps): JSX.Element => (
    <Layout
        orientation={'row'}
        style={{
            ...style,
            alignItems: 'center',
            width: '100%',
            fontFamily: "'IBM Plex Sans', sans-serif",
        }}
    >
        <div
            style={{
                backgroundColor: '#BEBEBE',
                width: '100%',
                height: '2px',
            }}
        />
        <div
            style={{
                padding: '0 8px',
                color: '#BEBEBE',
                verticalAlign: 'center',
            }}
        >
            {children}
        </div>
        <div
            style={{
                backgroundColor: '#BEBEBE',
                width: '100%',
                height: '2px',
            }}
        />
    </Layout>
);

export default Divider;
