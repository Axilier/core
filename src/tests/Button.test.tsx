import React from 'react';
import { mount } from 'enzyme';
import { Button } from '../index';
import { Google } from '../components/icons';

const onClick = jest.fn();
const buttonContainer = mount(<Button onClick={onClick}>{'click me'}</Button>);

describe('when the button is used in an application', () => {
    it('should render the contained button correctly', () => {
        expect(buttonContainer).toMatchSnapshot();
    });
    it('should contain a button element', () => {
        expect(buttonContainer.length).toBe(1);
        expect(buttonContainer.find('#button').type()).toEqual('button');
    });
    it('should have `click me` in the button', () => {
        expect(buttonContainer.find('#button').text()).toEqual('click me');
    });
    describe('when the `children` is less than 14 characters long', () => {
        it('should have a vertical padding of 5px', () => {
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('padding', '5px 20px');
        });
    });
    describe('when the `children` is more than 14 characters long', () => {
        beforeAll(() =>
            buttonContainer.setProps({
                children: 'This is a test for string larger than 14 char',
            }),
        );
        it('should have a vertical padding of 5px', () => {
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('padding', '10px 20px');
        });
        afterAll(() => buttonContainer.setProps({ children: 'click me' }));
    });
    describe('when the btnStyles prop is populated', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnStyles: { height: '200px' } });
        });
        it('should apply styles passed in the `btnStyles` prop directly to the button and override existing ones', () => {
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('height', '200px');
        });
        afterAll(() => {
            buttonContainer.setProps({ btnStyles: undefined });
        });
    });
    describe('when the btnIcon prop is populated', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnIcon: <Google /> });
        });
        it('should render the icon before the other children', () => {
            expect(
                buttonContainer
                    .childAt(0)
                    .childAt(0)
                    .childAt(0)
                    .childAt(0)
                    .equals(<Google />),
            ).toBe(true);
        });
        describe('when the `btnIconBackgroundColor` prop is defined', () => {
            beforeAll(() => {
                buttonContainer.setProps({ btnIconBackgroundColor: 'red' });
            });
            it('should render the icon with the correct direct styles', () => {
                expect(
                    buttonContainer.find('#button').childAt(0).props().style,
                ).toEqual({
                    backgroundColor: 'red',
                });
            });
            afterAll(() => {
                buttonContainer.setProps({ btnIconBackgroundColor: undefined });
            });
        });
        describe('when the `btnIconBackgroundColor` is not defined', () => {
            beforeAll(() => {
                buttonContainer.setProps({ btnIconBackgroundColor: null });
            });
            it('should show a white background for the icon', () => {
                expect(
                    buttonContainer.find('#button').childAt(0).props().style,
                ).toEqual({
                    backgroundColor: 'white',
                });
            });
        });
        afterAll(() => {
            buttonContainer.setProps({ btnIcon: undefined });
        });
    });
    describe('when the `btnIcon` prop is not populated', () => {
        it('should not render the icon', () => {
            expect(buttonContainer.find('#button').children()).toHaveLength(1);
        });
    });
    describe('when the `btnType` is set', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnType: 'submit' });
        });
        it('should be of type submit', () => {
            expect(buttonContainer.find('#button').props()).toHaveProperty(
                'type',
                'submit',
            );
        });
    });
    describe('when the `btnType` is unset', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnType: null });
        });
        it('should be of type submit', () => {
            expect(buttonContainer.find('#button').props()).toHaveProperty(
                'type',
                'button',
            );
        });
    });
    describe('when the button is disabled', () => {
        beforeAll(() => buttonContainer.setProps({ disabled: true }));
        it('should have a filter element on top of the main button', () => {
            expect(buttonContainer.find('#filter').exists()).toBeTruthy();
        });
        afterAll(() => buttonContainer.setProps({ disabled: false }));
    });
    describe('when the button is enabled', () => {
        it('should not have a filter element on top of the main button', () => {
            expect(buttonContainer.find('#filter').exists()).toBeFalsy();
        });
    });
    describe('when `styles` are provided', () => {
        beforeAll(() =>
            buttonContainer.setProps({ style: { margin: '20px' } }),
        );
        it('should pass those styles onto the container of the button', () => {
            expect(buttonContainer.props().style).toHaveProperty(
                'margin',
                '20px',
            );
        });
    });
});
describe('when the button is the contained variant', () => {
    beforeAll(() => {
        buttonContainer.setProps({ variant: 'contained' });
    });
    describe('when the button color is primary', () => {
        it('should have the correct styles for a default contained button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: '#057AFF',
                color: 'white',
                border: 'solid 2px transparent',
                padding: '5px 20px',
            });
        });
        it('should have the correct styles for a default contained hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'white',
                color: '#057AFF',
                border: 'solid 2px #057AFF',
                padding: '5px 20px',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
    describe('when the button color is red', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnColor: 'red' });
        });
        it('should have the correct styles for a set coloured button', () => {
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('backgroundColor', 'red');
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('color', 'white');
        });
        it('should have the correct styles for a set coloured hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'white',
                color: 'red',
                border: 'solid 2px red',
                padding: '5px 20px',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
    describe('when the button color is secondary', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnColor: 'secondary' });
        });
        it('should have the correct styles for a secondary contained button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'black',
                color: 'white',
                border: 'solid 2px transparent',

                padding: '5px 20px',
            });
        });
        afterAll(() => {
            buttonContainer.setProps({ btnColor: 'primary' });
        });
    });
});
describe('when the button is the outlined variant', () => {
    beforeAll(() => {
        buttonContainer.setProps({
            variant: 'outlined',
            btnColor: 'primary',
        });
    });
    describe('when the button color is primary', () => {
        it('should have the correct styles for a default button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'white',
                color: '#057AFF',
                border: 'solid 2px #057AFF',

                padding: '5px 20px',
            });
        });
        it('should have the correct styles for a default hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: '#057AFF',
                color: 'white',
                border: 'solid 2px #057AFF',

                padding: '5px 20px',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
    describe('when the button color is red', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnColor: 'red' });
        });
        it('should have the correct styles for a set coloured button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'white',
                color: 'red',
                border: 'solid 2px red',

                padding: '5px 20px',
            });
        });
        it('should have the correct styles for a set coloured hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'red',
                color: 'white',
                border: 'solid 2px red',

                padding: '5px 20px',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
});
describe('when the button is the text variant', () => {
    beforeAll(() => {
        buttonContainer.setProps({
            variant: 'text',
            btnColor: 'primary',
        });
    });
    describe('when the button color is primary', () => {
        it('should have the correct styles for a default button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'transparent',
                color: '#057AFF',
                border: 'none',
                padding: 'unset',
            });
        });
        it('should have the correct styles for a default hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'transparent',
                color: '#057AFF',
                border: 'none',
                padding: 'unset',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
    describe('when the button color is red', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnColor: 'red' });
        });
        it('should have the correct styles for a default button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'transparent',
                color: 'red',
                border: 'none',

                padding: 'unset',
            });
        });
        it('should have the correct styles for a default hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'transparent',
                color: 'red',
                border: 'none',
                padding: 'unset',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
});
describe('when the button`s variant is undefined', () => {
    beforeAll(() => {
        buttonContainer.setProps({ variant: null, btnColor: 'primary' });
    });
    describe('when the button color is primary', () => {
        it('should have the correct styles for a default button', () => {
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: '#057AFF',
                color: 'white',
                border: 'solid 2px transparent',
                padding: '5px 20px',
            });
        });
        it('should have the correct styles for a default hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'white',
                color: '#057AFF',
                border: 'solid 2px #057AFF',
                padding: '5px 20px',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
    describe('when the button color is red', () => {
        beforeAll(() => {
            buttonContainer.setProps({ btnColor: 'red' });
        });
        it('should have the correct styles for a set coloured button', () => {
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('backgroundColor', 'red');
            expect(
                buttonContainer.find('#button').props().style,
            ).toHaveProperty('color', 'white');
        });
        it('should have the correct styles for a set coloured hovered button', () => {
            buttonContainer.find('#button').simulate('mouseenter');
            expect(buttonContainer.find('#button').props().style).toEqual({
                backgroundColor: 'white',
                color: 'red',
                border: 'solid 2px red',
                padding: '5px 20px',
            });
            buttonContainer.find('#button').simulate('mouseleave');
        });
    });
});
