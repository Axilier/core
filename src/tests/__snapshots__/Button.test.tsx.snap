// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the button is used in an application should render the contained button correctly 1`] = `
<Button
  onClick={[MockFunction]}
>
  <div
    className="buttonContainer undefined"
  >
    <button
      className="button"
      id="button"
      onClick={[MockFunction]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "backgroundColor": "#057AFF",
          "border": "solid 2px transparent",
          "color": "white",
          "padding": "5px 20px",
        }
      }
      type="button"
    >
      click me
    </button>
  </div>
</Button>
`;
